src/Fixer/ArrayNotation/AlphabeticalArrayKeySortFixer.php

```diff
@@ -0,0 +1,312 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ArrayNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Abdurrahman Uymaz <abdurrahman.uymaz@mobian.global>
+ * @author Lars Grevelink <lars.grevelink@mobian.global>
+ * @author Leander Philippo <leander.philippo@mobian.global>
+ */
+final class AlphabeticalArrayKeySortFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Sorts keyed arrays alphabetically.',
+            [
+                new CodeSample(
+                    "<?php\n\$sample = array('b' => '2', 'a' => '1', 'd' => '5');\n"
+                ),
+                new CodeSample(
+                    "<?php\n\$sample = ['b' => '2', 'a' => '1', 'd' => '5'];\n",
+                    []
+                ),
+                new CodeSample(
+                    "<?php\n\$sample = ['b' => '2', 'a' => '1', foo() => 'bar', 'd' => '5'];\n",
+                    ['sort_special_key_mode' => 'special_case_on_bottom']
+                ),
+                new CodeSample(
+                    "<?php\n\$sample = ['b' => '2', 'a' => '1', foo() => 'bar', 'd' => '5'];\n",
+                    ['sort_special_key_mode' => 'special_case_on_top']
+                ),
+            ],
+            'Alphabetically sorts any keyed array on its key values.',
```

Comment:
The description must explain [reasoning why/when this fixer should be used](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7071/files#r1264715309) and this clearly is not doing that.

Reply:
Most of the fixers which were already in `master` didn't have one or copied the summary. Since it was made required between the updates I copied the idea of which had one. Thanks for the thread, I have no clue what is expected to add still but I will attempt to draw something up.

Reply:
Expectation was reverted, so we can close this discussion anyway.